File: test/fixtures/java-rest-controllers/spring-entity-controller.java
package com.example;

import com.example.dto.EntityDto;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/entity")
public class EntityController {
    @PutMapping("/{id}")
    public ResponseEntity<EntityDto> update(@PathVariable Long id, @RequestBody EntityDto dto) { }
}

File: test/fixtures/kotlin-rest-controllers/spring-entity-controller.kt
package com.example

import com.example.dto.EntityDto
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/entity")
class EntityController {
    @PutMapping("/{id}")
    fun update(@PathVariable id: Long, @RequestBody dto: EntityDto): ResponseEntity<EntityDto> { }
}
